file_test.go

```diff
@@ -0,0 +1,89 @@
+package main
+
+import (
+	"os"
+	"testing"
+	"path/filepath"
+	"io/ioutil"
+)
+
+func TestDownloadZipFile(t *testing.T) {
+	cases := []struct {
+		repoOwner   string
+		repoName    string
+		gitTag      string
+		githubToken string
+	}{
+		{"gruntwork-io", "fetch-test-public", "v0.0.1", ""},
+		{"gruntwork-io", "fetch-test-private", "v0.0.2", os.Getenv("GITHUB_OAUTH_TOKEN")},
```

Comment:
Add an explicit check at the top of this test file that checks if `GITHUB_OAUTH_TOKEN` is set and fails the test with a clear error message if it's not. Otherwise, someone running this locally will be confused why the test fails right after checkout.